### DISCUSSION TITLE
err: Error: ETXTBSY: text file is busy, copyfile `query-engine-debian-openssl-1.1.x`

### DISCUSSION THREAD

**Discussion by meotimdihia** [prisma/prisma#6938] [Q&A] [answered]
How to install a new version of Prisma without killing the remaining `query-engine` process is running?
I always have to kill the Prisma running process to deploy the project with the new version of Prisma.

`err: Error: ETXTBSY: text file is busy, copyfile '/home/ubuntu/sources/node_modules/prisma/query-engine-debian-openssl-1.1.x' -> '/home/ubuntu/sources/node_modules/.prisma/client/query-engine-debian-openssl-1.1.x'`

Do we have solutions for the problem?

**Comment by janpio** [comment, marked as answer]
Wow, that is an interesting question. 

What is the downside of killing the running Prisma QE process that makes you want to avoid that?

I do not think we have any solution here yet - but certainly could put the query engine in different locations per version to avoid this problem (for example).


Can you please open a feature request issue? We should definitely look at that.

**Reply by meotimdihia** [reply]
Hi, I don't know about the downside of killing the running `query-engine-debian-openssl-1.1.x`.
Doesn't it mean the running query would be killed? and I want to avoid it
My website has a lot of long-running processes. I want to ask If we are currently don't have any other way.
If we don't have one, then I will continue to kill the processes for updating. No big problems happened for now.

**Reply by janpio** [reply]
> Doesn't it mean the running query would be killed? and I want to avoid it

That is a good point. You can probably shut it down properly if you shut down your server, which would also remove the lock on the file and let you update properly. But of course not while your server is running.
